Sources/TensorFlow/Core/DifferentialOperators.swift

```diff
@@ -29,7 +29,10 @@ public extension Differentiable {
         in f: @differentiable (Self) -> Tensor<R>
     ) -> (value: Tensor<R>, gradient: TangentVector) {
         let (y, pb) = self.valueWithPullback(in: f)
-        precondition(y.rank == 0)
+        precondition(y.rank == 0, """
+            The function being differentiated produced a tensor with shape \(y.shape).
```

Comment:
It is not common for precondition messages to have multiple lines. Please add a space followed by a backslash at the end of the first line in the message. This will make the printed message become one line.

Reply:
SGTM, thanks! Done!